src/library/components/resource-lightbox.js

```diff
@@ -31,18 +38,29 @@ var ResourceLightbox = Component({
     portalObjectHelpers.processResource(resource)
 
     this.titleSuffix = document.title.split('|')[1] || ''
+    this.setState({
+      openAssign: false
+    })
     this.replaceResource(resource)
   },
 
   componentDidMount: function () {
+    if (this.state.openAssign) {
+      jQuery('#assign-button')[0].click()
+    }
     jQuery('.portal-pages-resource-lightbox-background, .portal-pages-resource-lightbox-container').fadeIn()
   },
 
   componentWillUnmount: function () {
-    document.title = this.props.savedTitle
+    document.title = this.state.savedTitle
     try {
-      window.history.replaceState({}, document.title, this.props.savedUrl)
+      if (this.state.parentPage !== '/' && this.state.parentPage !== PortalPages.initialPath) {
+        window.location.href = this.state.parentPage
+      } else {
+        window.history.replaceState({}, document.title, this.state.parentPage)
+      }
     } catch (e) {}
+
     jQuery('html, body').css('overflow', 'auto')
     jQuery('.home-page-content').removeClass('blurred')
```

Comment:
@scytacki I made this change. I also added a line that makes the page look blank immediately after the lightbox closes so that the reload is even less confusing/noticeable.